digestwriter/consumer.go

```diff
@@ -244,8 +252,44 @@ func extractDigestsFromMessage(workload Workload) (digests []string) {
 	return
 }
 
+// isMessageInGzipFormat will check if the format of the message is gzip if it is it will return true if not it will return false
+func isMessageInGzipFormat(messageValue []byte) bool {
+	if messageValue == nil {
+		return false
+	}
+	if len(messageValue) < 2 {
+		return false
+	}
+	// Checking for first 2 bytes in gzip instance witch are 31 and 139
+	if messageValue[0] == gzip1 && messageValue[1] == gzip2 {
+		return true
+	}
+	return false
+
+}
```

Comment:
You don't need to do this since its already called here https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/compress/gzip/gunzip.go;drc=b2b8872c876201eac2d0707276c6999ff3eb185e;l=185 :slightly_smiling_face:

Reply:
If source do not provide flag for compression, then maybe its okay to check it this way and don't worry about above comment :slightly_smiling_face:

Reply:
I wanted to check it before creating the reader to save time so should i change it or let it be as it is ?

Reply:
If we don't know if message is compressed or not then it could be a good idea and faster indeed. It would be best if you would get from somewhere real world compressed message and add it to unit tests, to make sure it works and it indeed starts with `ID1` and `ID2` gip headers.

Reply:
I allready added some test that compress the messages and then it checks them is they are compressed and after that it parse them for example  TestDigestCompressedMessageParse in consumer_test.go

Reply:
okay